packages/mongodb-rag-core/src/findContent/DefaultFindContent.ts

```diff
@@ -28,7 +28,7 @@ export const makeDefaultFindContent = ({
     let content = await store.findNearestNeighbors(embedding, {
       ...findNearestNeighborsOptions,
       filter: filters,
-      // TODO: need to add logic to pass limit to findNearestNeighbors
+      ...(limit ? { k: limit }: {})
     });
```

Comment:
This is necessary to pass tests - `k` must be a number (not even `undefined` if present)

Reply:
ah interesting. i think this is b/c of a previous bug w/ the mongodb node.js driver where it'd throw if you passed `[key]: undefined` for certain properties. probably no longer necessary but i'm fine keeping as is

Reply:
chatbot lore: the above node.js driver bug actually caused the only major downtime incident that the chatbot has ever had